clang/test/SemaSYCL/unnamed-kernel.cpp

```diff
@@ -89,20 +101,31 @@ struct MyWrapper {
 
     using InvalidAlias1 = InvalidKernelName5;
 #ifndef __SYCL_UNNAMED_LAMBDA__
-    // expected-error@Inputs/sycl.hpp:220 {{kernel needs to have a globally-visible name}}
-    // expected-note@24 {{InvalidKernelName5 declared here}}
+    // expected-error@Inputs/sycl.hpp:220 {{invalid type 'MyWrapper::InvalidKernelName5' used in kernel name. Kernel name should be globally-visible}}
+    // expected-note@Inputs/sycl.hpp:220 {{Invalid kernel name is 'namespace1::KernelName<MyWrapper::InvalidKernelName5>'}}
+    // expected-note@30 {{InvalidKernelName5 declared here}}
     // expected-note@+3{{in instantiation of function template specialization}}
 #endif
     q.submit([&](cl::sycl::handler &h) {
       h.single_task<namespace1::KernelName<InvalidAlias1>>([] {});
     });
+#ifndef __SYCL_UNNAMED_LAMBDA__
+    // expected-error@Inputs/sycl.hpp:220 {{invalid type 'InvalidKernelName1' used in kernel name. Kernel name should be globally-visible}}
+    // expected-note@Inputs/sycl.hpp:220 {{Invalid kernel name is 'Templated_kernel_name2<Templated_kernel_name<InvalidKernelName1>>'}}
```

Comment:
Possible suggestion for diagnostic message to keep it shorter:
error: `'Templated_kernel_name2<Templated_kernel_name<InvalidKernelName1>>' is invalid kernel name`
note: `'InvalidKernelName1' type is not globally-visible`

Reply:
Done